sandy/fy.py

```diff
@@ -422,23 +422,32 @@ def apply_qmatrix(self, zam, energy, decay_data):
         >>> decay_minimal = sandy.get_endf6_file("jeff_33", 'decay', zam)
         >>> decay_fytest = sandy.DecayData.from_endf6(decay_minimal)
         >>> npfy = Fy(minimal_fytest_2)
-        >>> npfy_pert = npfy.apply_bmatrix(942390, 5.00000e+05, decay_fytest)
-        >>> diff = npfy_pert.data[npfy_pert.data.FY != npfy.data.FY].FY
-        >>> comp = npfy_pert.data.query('ZAM==942390 & MT==459 & E==500e3').squeeze().FY
-        >>> assert comp.values.all() == diff.values.all()
-
-        """
-        new_data = self.data.copy()
+        >>> npfy_pert = npfy.apply_qmatrix(942390, 5.00000e+05, decay_fytest)
+        >>> npfy_pert.data[npfy_pert.data.MT == 459]
+             MAT	 MT	   ZAM	   ZAP	          E	         FY	        DFY
+        3	9437	459	942390	591480	5.00000e+05	1.00000e-01	0.00000e+00
+        4	9437	459	942390	591481	5.00000e+05	2.00000e-01	0.00000e+00
+        5	9437	459	942390	601480	5.00000e+05	6.00000e-01	0.00000e+00
+        6	9437	459	942390	621480	5.00000e+05	6.00000e-01	0.00000e+00
+        """
+        # Obtain the data:
+        data = self.data.copy()
         conditions = {'ZAM': zam, 'MT': 454, "E": energy}
-        fy_data = self._filters(conditions).data.set_index('ZAP')['FY']
-        index = fy_data.index
+        fy_data = self._filters(conditions).data
+        mat = fy_data.iloc[0, 0]
+        fy_data = fy_data.set_index('ZAP')['FY']
         Q = decay_data.get_qmatrix()
+        # Put the data in a approppiate format:
+        mask = (data.ZAM == zam) & (data.MT == 459) & (data.E == energy)
+        data = pd.concat([data, data.loc[mask]]).drop_duplicates(keep=False)
         fy_data = fy_data.reindex(Q.columns).fillna(0)
         # Apply qmatrix
-        fy_calc = Q.dot(fy_data).loc[index]
-        mask = (new_data.ZAM == zam) & (new_data.MT == 459) & (new_data.E == energy)
-        new_data.loc[mask, 'FY'] = fy_calc.values
-        return self.__class__(new_data)
+        fy_calc_values = Q.dot(fy_data).rename('FY').to_frame()
```

Comment:
better call it `cfy_calc_values`

Reply:
put lines 445 and 446 together:
`cfy_calc_values = Q.dot(fy_data).rename('FY').reset_index().rename(columns={'DAUGHTER': 'ZAP'}`

if you use `reset_index` you don't need to use `to_frame`

Reply:
But reset_index only work in a dataFrame and you obtain a pd.Series from the multiplication

Reply:
https://pandas.pydata.org/docs/reference/api/pandas.Series.reset_index.html

Reply:
This is also still open...why not `reset_index` instead of `to_frame().reset_index`?
It doesn't work?

Reply:
@AitorBengoechea , this was not resolved? is it now?

Reply:
Yes, now it returns in the output the 3850 nucleus that are in the rows of the Q matrix.